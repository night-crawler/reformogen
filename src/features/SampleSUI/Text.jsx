import React from 'react';
import PropTypes from 'prop-types';
import { Form, Checkbox, Segment } from 'semantic-ui-react';

export class Text extends React.Component {
  static propTypes = {
    uuid: PropTypes.string,
    data: PropTypes.any,
  };

  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Segment as={ Form } className='Text Block' color='brown'>
        { this.props.uuid } - { this.props.data }
        <Checkbox
          label='EN > RU'
          toggle={ true }
        />
      </Segment>
    );
  }
}
