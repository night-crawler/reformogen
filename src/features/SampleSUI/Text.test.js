/* eslint-disable react/react-in-jsx-scope,no-undef */
import { Text } from './Text';

describe('<Text />', () => {
  it('should render Text component', () => {
    const wrapper = shallow(
      <Text data='test' uuid='uuid' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
