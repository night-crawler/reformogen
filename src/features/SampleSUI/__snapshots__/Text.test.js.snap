// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should render Text component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Text
    data="test"
    uuid="uuid"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": [Function],
      "children": Array [
        "uuid",
        " - ",
        "test",
        <Checkbox
          label="EN > RU"
          toggle={true}
          type="checkbox"
        />,
      ],
      "className": "Text Block",
      "color": "brown",
    },
    "ref": null,
    "rendered": Array [
      "uuid",
      " - ",
      "test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "label": "EN > RU",
          "toggle": true,
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "children": Array [
          "uuid",
          " - ",
          "test",
          <Checkbox
            label="EN > RU"
            toggle={true}
            type="checkbox"
          />,
        ],
        "className": "Text Block",
        "color": "brown",
      },
      "ref": null,
      "rendered": Array [
        "uuid",
        " - ",
        "test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "EN > RU",
            "toggle": true,
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
